tic-tac-toe/src/Board.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {useContext} from 'react';
+import Square from './Square';
+import BoardState from './contexts/BoardContext';
+import declareWinner from './contexts/declareWinner';
+
+const Board = (props) => {
+  const { turn, setTurn, boardState, setBoardState, history, setHistory} = useContext(BoardState);
+
+  const winner = declareWinner(boardState);
+  let status;
+  if (winner) {
+    status = `The winner is ${winner}!`;
+  }
+
+  function handleClick(num) {
+    if (!declareWinner(boardState) && !boardState[num]) {
+      const updatedHistory = [...history];
+      const updatedBoard = [...boardState];
+      updatedBoard[num] = turn;
+      updatedHistory.push(updatedBoard);
+      setBoardState(updatedBoard);
+      setHistory(updatedHistory);
+      setTurn((turn === 'O') ? 'X' : 'O');
+    }
+  }
+
+  function stepTo(step) {
+    const boardInHistory = [...history];
+    setBoardState(boardInHistory[step - 1] || []);
+    setHistory(boardInHistory[step - 1] || []);
+    setTurn((step % 2 === 0) ? 'X' : 'O');
+  }
+
+  const playerMoves = history.map((step, move) => {
+    const desc = move ?
```

Comment:
:thumbsup: on configuring this copy - stuff like this can be a bit of a pain. Also, great work on using`map` here to easily render the elements!